### PR TITLE
Refactor Notifications to use Vue, and add clear all option

### PR DESCRIPTION
**Pull request by deeptailor**
This PR fixes issue #3006, and refactors angular out of notifications

### CODE REVIEW

File: platform/commonUI/notification/src/NotificationIndicatorController.js [hunk 1]
```diff
@@ -48,6 +48,7 @@ define(
                         notification.model.primaryOption = {
                             label: 'Dismiss',
                             callback: () => {
+                                console.log('notification dismissed');
```
Comment (nikhilmandlik) [inline platform/commonUI/notification/src/NotificationIndicatorController.js:51]: same

File: platform/commonUI/notification/src/NotificationIndicatorController.js [hunk 2]
```diff
@@ -61,7 +62,16 @@ define(
                         title: "Messages",
                         //Launch the message list dialog with the models
                         // from the notifications
-                        messages: notificationsList
+                        messages: notificationsList,
+                        topBarButton: {
+                            onClick: () => {
+                                console.log('clicked');
```
Comment (nikhilmandlik) [inline platform/commonUI/notification/src/NotificationIndicatorController.js:68]: same

File: platform/commonUI/dialog/res/templates/overlay-message-list.html [hunk 3]
```diff
@@ -6,6 +6,12 @@
                     ng-show="ngModel.dialog.messages.length > 1 ||
                             ngModel.dialog.messages.length == 0">s</span>
             </div>
+            <button
+                ng-if="ngModel.dialog.topBarButton"
+                class="c-button c-button--major"
+                ng-click="ngModel.topBarButton.onClick">
+                {{console.log(ng-model.topBarButton)}}
```
Comment (nikhilmandlik) [inline platform/commonUI/dialog/res/templates/overlay-message-list.html:13]: console.log ?

File: src/plugins/notificationIndicator/components/NotificationsList.vue [hunk 4]
```diff
@@ -0,0 +1,62 @@
+<template>
+<div class="t-message-list c-overlay__contents">
+    <div class="c-overlay__top-bar">
+        <div class="c-overlay__dialog-title">Notifications</div>
+        <div class="c-overlay__dialog-hint">
+            Displaying {{ notifications.length }} notification<span v-show="notifications.length > 1 || notifications.length === 0">s</span>
```
Comment (nikhilmandlik) [inline src/plugins/notificationIndicator/components/NotificationsList.vue:6]: same, may be better if its a computed prop

**Comment by deeptailor** [comment]
@nikhilmandlik Made changes that you requested. Also added a test and removed files that were not being used

**Comment by deeptailor** [comment]
## Author Checklist
Changes address original issue? Yes
Unit tests included and/or updated with changes? Yes
Command line build passes? Yes
Changes have been smoke-tested? Yes
Testing instructions included? Yes

**Comment by deeptailor** [comment]
Testing instructions:

1. Install 'example.notifications' plugin on line 41 of index.html
2. Using the example notifications indicator, generate a bunch of alerts, error etc messages
3. Click on the new notifications indicator to see the notifications
4. Click on the clear all notifications and verify the notifications are cleared

**Comment by jvigliotta** [comment]
Verified Fixed

**Pull request merged**